include/dkm.hpp

```diff
@@ -207,6 +207,21 @@ std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> kmeans_lloyd(
 	return std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>>(means, clusters);
 }
 
+
+template <typename T, size_t N>
+size_t predict(std::vector<std::array<T, N>>& centroids, const std::array<T, N> query) {
+    T min = std::numeric_limits<T>::max();
+    size_t index = 0;
```

Comment:
Rather than using `std::numeric_limits<T>::max();` and `0`, why not use the first element of `centroids` and start iterating from `i = 1`? Yes, the existing code is unlikely to accidentally return these initial values, but this change would make it completely impossible.